src/client/dfuse/dfuse_core.c

```diff
@@ -21,9 +21,48 @@
  * portions thereof marked with this legend must also reproduce the markings.
  */
 
+#include <pthread.h>
+
 #include "dfuse_common.h"
 #include "dfuse.h"
 
+static void *
+dfuse_progress_thread(void *arg)
+{
+	struct dfuse_projection_info *fs_handle = arg;
+	int rc;
+	daos_event_t *dev;
+	struct dfuse_event *ev;
+
+	while (1) {
+
+		errno = 0;
+		rc = sem_wait(&fs_handle->dpi_sem);
+		if (rc != 0) {
+			rc = errno;
```

Comment:
should we check to make sure the error isn't EINTR ?  In the example on the man page, it simply restarts the wait in such cases

Reply:
Good point.  The error handling here is always going to be tricky but I can think of a lot of reasons that EINTR would be returned here.  It would probably mean that a semaphore event was "lost" so the code would always fall through here and block in the eq_poll() function below, so I suspect everything would continue to work but at 100% CPU rather than 0%.  I'll see if I can reproduce this tomorrow.

Reply:
I wasn't able to reproduce this, even by sending a signal to the specific thread however I've added some code to handle it.